src/views/catalog/wizard/tabs/scripts/components/SSHKey.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Trans } from 'react-i18next';
+
+import { useWizardVMContext } from '@catalog/utils/WizardVMContext';
+import { WizardDescriptionItem } from '@catalog/wizard/components/WizardDescriptionItem';
+import { SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { AuthorizedSSHKeyModal } from '@kubevirt-utils/components/AuthorizedSSHKeyModal/AuthorizedSSHKeyModal';
+import { addSecretToVM } from '@kubevirt-utils/components/CloudinitModal/utils/cloudinit-utils';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Stack, Text, TextVariants } from '@patternfly/react-core';
+
+import { changeSecretKey, removeSSHKeyObject, updateSSHKeyObject } from './sshkey-utils';
+
+const SSHKey: React.FC = () => {
+  const { vm, updateVM, tabsData, updateTabsData } = useWizardVMContext();
+  const { t } = useKubevirtTranslation();
+  const { createModal } = useModal();
+
+  const vmAttachedSecretName = vm?.spec?.template?.spec?.accessCredentials?.find(
+    (ac) => ac?.sshPublicKey?.source?.secret?.secretName,
+  )?.sshPublicKey?.source?.secret?.secretName;
+
+  const sshSecretObject = tabsData?.additionalObjects?.find(
+    (object) => object.kind === SecretModel.kind && object?.metadata?.name === vmAttachedSecretName,
+  );
+  const secretKey = sshSecretObject?.data?.key && atob(sshSecretObject?.data?.key);
+
+  const onSSHChange = (secretName: string, sshKey?: string) => {
+    if (!sshKey && !secretName) {
+      return updateVM((vmDraft) => {
+        vmDraft.spec.template.spec.accessCredentials = null;
+      });
+    }
+
+    if (sshKey) {
+      return sshSecretObject
+        ? Promise.resolve(changeSecretKey(updateTabsData, sshKey, vmAttachedSecretName))
+        : updateSSHKeyObject(vm, updateVM, updateTabsData, sshKey);
+    } else {
+      removeSSHKeyObject(updateTabsData, vmAttachedSecretName);
+    }
+
+    if (secretName) {
```

Comment:
if you move this to line 35, u will be able to remove the else statement in line 40

Reply:
Great thanks!!